master/pkg/searcher/hyperparameters.go

```diff
@@ -10,6 +10,17 @@ import (
 
 type hparamSample map[string]interface{}
 
+func (h hparamSample) GlobalBatchSize() int {
+	// If the hyperparameters.global_batch_size is configured as a const hyperparameter,
+	// we infer its type to be a float but in some cases, its type can be specified and an
+	// int is also valid.
```

Comment:
This isn't because someone chose a float but because when golang sees a numeric value and is told to deserialize it into an `interface{}` like in:
```
// ConstHyperparameter is a constant.
type ConstHyperparameter struct {
	Val interface{} `json:"val"`
}
```
it _always_ chooses a float64:

https://play.golang.org/p/ridKntaeRMp